bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,344 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dominoswiss.internal;
+
+import static org.openhab.binding.dominoswiss.internal.dominoswissBindingConstants.*;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EgateHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Frieso Aeschbacher - Initial contribution
+ */
+@NonNullByDefault
+public class eGateHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(eGateHandler.class);
+    private @Nullable Socket egateSocket;
+
+    private int port;
+    private @Nullable String host;
+    private static final int SOCKET_TIMEOUT = 250;
+    private final Object lock = new Object();
+    private @Nullable BufferedWriter writer;
+    private @Nullable BufferedReader reader;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable Map<String, ThingUID> registeredBlinds;
+    private final ScheduledExecutorService scheduler;
+
+    public eGateHandler(Bridge thing) {
+        super(thing);
+        registeredBlinds = new HashMap<String, ThingUID>();
+        scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
You could do the following to create a thread that is correctly configured and named, according the coding guidelines.
```suggestion
        scheduler = Executors.newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
```